src/hooks/useLifecycleLogger.ts

```diff
@@ -0,0 +1,42 @@
+import type { EffectCallback } from 'react';
+import { useRef, useEffect } from 'react';
+
+const useEffectOnce = (effect: EffectCallback) => {
```

Comment:
We already have these hooks right? Let's directly use them instead of creating them.